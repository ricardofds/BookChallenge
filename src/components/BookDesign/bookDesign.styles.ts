import { StyleSheet } from 'react-native';
import { COLORS, SIZES } from '../../constants/theme';

const bookDesign = StyleSheet.create({
  cardTouch: {
    flex: 1,
    marginRight: SIZES.radius,
    marginBottom: 20,
  },
  cardImage: {
    width: 180,
    height: 250,
    borderRadius: 20,
  },
  cardTitleView: {
    marginTop: SIZES.radius,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  cardTitleText: {
    marginLeft: 10,
    color: COLORS.black,
    fontSize: 12,
  },
});

export default bookDesign;
