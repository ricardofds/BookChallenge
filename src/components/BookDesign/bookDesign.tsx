import React from 'react';
import { Image, Text, TouchableOpacity, View } from 'react-native';

import styles from './bookDesign.styles';

interface IBookDesignProps {
  image?: string;
  name: string;
}

const imageDefault =
  'https://cdn.neemo.com.br/uploads/settings_webdelivery/logo/2609/image-not-found.jpg';

const BookDesign = ({ image = imageDefault, name }: IBookDesignProps) => {
  return (
    <TouchableOpacity style={styles.cardTouch}>
      <Image
        source={{
          uri: image,
        }}
        resizeMode="cover"
        style={styles.cardImage}
      />

      <View style={styles.cardTitleView}>
        <Text style={styles.cardTitleText}>{name}</Text>
      </View>
    </TouchableOpacity>
  );
};

export default BookDesign;
