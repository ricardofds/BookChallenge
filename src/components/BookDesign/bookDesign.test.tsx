import React from 'react';
import { render, screen, fireEvent, act } from '@testing-library/react-native';
import BookDesign, { imageDefault } from './bookDesign';

describe('Book Design Component', () => {
  it('should render correctly', () => {
    const pressMock = jest.fn();

    render(
      <BookDesign
        name="Teste unitário"
        image="https://upload.wikimedia.org/wikipedia/commons/thumb/a/a7/React-icon.svg/1200px-React-icon.svg.png"
        onPress={pressMock}
      />,
    );

    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('should render correctly without image', () => {
    const pressMock = jest.fn();

    render(<BookDesign name="Teste unitário" onPress={pressMock} />);

    const image = screen.getByTestId('book_design__image');

    expect(image.props.source.uri).toBe(imageDefault);
  });

  it('should render correctly with onPress', async () => {
    const pressMock = jest.fn();

    render(
      <BookDesign testID="unit" name="Teste unitário" onPress={pressMock} />,
    );

    await act(async () => {
      const buttonPress = screen.getByTestId('book_design__action__unit');

      fireEvent.press(buttonPress);
    });

    expect(pressMock).toBeCalled();
  });
});
