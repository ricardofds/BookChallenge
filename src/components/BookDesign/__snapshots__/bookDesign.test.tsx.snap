// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book Design Component should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "marginBottom": 20,
      "marginRight": 12,
      "opacity": 1,
      "width": 180,
    }
  }
  testID="book_design__action__undefined"
>
  <Image
    resizeMode="cover"
    source={
      Object {
        "uri": "https://upload.wikimedia.org/wikipedia/commons/thumb/a/a7/React-icon.svg/1200px-React-icon.svg.png",
      }
    }
    style={
      Object {
        "borderRadius": 20,
        "height": 250,
        "width": 180,
      }
    }
    testID="book_design__image"
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 12,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1E1B26",
          "fontSize": 12,
          "marginLeft": 10,
        }
      }
    >
      Teste unit√°rio
    </Text>
  </View>
</View>
`;
